pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -152,6 +154,33 @@ func (r *ReferenceAdapter) updateProjectID() error {
 	return r.kubeClient.Update(context.TODO(), r.projectReference)
 }
 
+// IsDeletionRequested checks the metadata.deletionTimestamp of ProjectReference instance, and returns if delete requested.
+// The controllers watching the ProjectReference use this as a signal to know when to execute the finalizer.
+func (r *ReferenceAdapter) IsDeletionRequested() bool {
+	return r.projectReference.DeletionTimestamp != nil
+}
+
+// AddPrjRefFinalizer parses the meta.Finalizers of ProjectReference instance and adds prjRefFinalizer if not found.
+// It returns a modification to the ProjectReference instance in the cluster for the controller to handle the reconciliation.
+func (r *ReferenceAdapter) AddPrjRefFinalizer() error {
+	if !clusterapi.Contains(r.projectReference.GetFinalizers(), prjRefFinalizer) {
+		r.projectReference.SetFinalizers(append(r.projectReference.GetFinalizers(), prjRefFinalizer))
+		return r.kubeClient.Update(context.TODO(), r.projectReference)
+	}
+	return nil
+}
+
+// DeletePrjRefFinalizer parses the meta.Finalizers of ProjectReference instance and removes prjRefFinalizer if found;
+// It returns a modification to the ProjectReference instance in the cluster for the controller to handle the reconciliation.
```

Comment:
This comment sounds like the function would return a ProjectReference, but it returns an error.